### PR TITLE
fix(openapi): error handling for incorrect project and version flags

### PR DESCRIPTION
**Pull request by kanadgupta**
## 🧰 Changes

One of several follow-ups to #410. In this one, we address [this unfinished test](https://github.com/readmeio/rdme/blob/7218933d50a08bbfc112c95617f1ef89484ed5aa/__tests__/cmds/openapi.test.js#L193) with some simple error handling and new tests.

Also removing a few redundant `.catch()` statements.

## 🧬 QA & Testing

If tests pass, we should be good! Here's a lil before and after with the scenarios in question:

**An invalid API key**:
```sh
➜  rdme git:(main) bin/rdme openapi ./__tests__/__fixtures__/ref-oas/petstore.json --key="invalid"

versionList.map is not a function

➜  rdme git:(fix/error-if-version-does-not-exist) ✗ bin/rdme openapi ./__tests__/__fixtures__/ref-oas/petstore.json --key="invalid"

We couldn't find your API key.

If you need help, email support@readme.io and include the following link to your API log: 'https://docs.readme.com/logs/xxx-xxx'.

```

**A valid API key but a version input that doesn't exist on the project**:
```sh
➜  rdme git:(main) bin/rdme openapi ./__tests__/__fixtures__/ref-oas/petstore.json --key="VALID_API_KEY" --version="v1000"

# In this case, it'll continue as normal and just ignore the flag and use project's default version, which is misleading 😬 

➜  rdme git:(fix/error-if-version-does-not-exist) bin/rdme openapi ./__tests__/__fixtures__/ref-oas/petstore.json --key="VALID_API_KEY" --version="v1000"

The version you specified (v1000) doesn't match any of the existing versions (1-hidden, 1-hidden-hidden-super-super-hidden, 1.0) in ReadMe.

If you need help, email support@readme.io and include the following link to your API log: 'https://docs.readme.com/logs/xxx-xxx'.
```

### CODE REVIEW

File: src/cmds/openapi.js [hunk 1]
```diff
@@ -118,17 +118,10 @@ exports.run = async function (opts) {
...
+      bundledSpec = JSON.stringify(res);
     });
-    await oas
-      .bundle()
-      .then(res => {
-        bundledSpec = JSON.stringify(res);
-      })
-      .catch(err => {
-        return Promise.reject(err);
-      });
```
Comment (kanadgupta) [inline src/cmds/openapi.js:131]: All the changes in this file are unrelated to the fix, these are just minor refactors. These `.catch(err => Promise.reject(err));` statements are redundant since we'd be rejecting the promise and handling that in `bin/rdme` anyways.

File: src/lib/versionSelect.js [hunk 2]
```diff
@@ -12,14 +13,14 @@ async function getProjectVersion(versionFlag, key, allowNewVersion) {
         method: 'get',
         headers: cleanHeaders(key),
       })
-        .then(res => res.json())
+        .then(res => handleRes(res))
```
Comment (kanadgupta) [inline src/lib/versionSelect.js:16]: Gotta love this little response handler that @rahulhegdee wrote for us 🤩

**Review by erunion — APPROVED** [review]
